### DISCUSSION TITLE
[IntelliSense] Underline valid TaildwindCSS classes

### DISCUSSION THREAD

**Discussion by HummingMind** [tailwindlabs/tailwindcss#11957] [Ideas]
Hello,
If possible, please add a feature to the VSCode extension to underline valid TailwindCSS classes. Similar to UnoCSS:

![image](https://github.com/tailwindlabs/tailwindcss/assets/87399963/1de65457-2bac-4c7f-8726-0f8dc75e9f9a)

It's extremely helpdful to see right away if a class name maybe misspelled or if there may be a custom class on a particular html element.

Thank you!

**Comment by feledori** [comment]
Just wanted to open up a discussion about this! This would be such a nice addition it really helps distinguish tailwind classes from other classes as well as helping find wrongly spelled classes!

**Comment by ackvf** [comment]
I remember the pain when you tried to debug why a rule was not applying and having to mouse-hover all classes to see how and if they even translated at all.

I haven't realized how much I want it until I got used to it on a project with UnoCSS.


![image](https://github.com/user-attachments/assets/1d299ed6-64d1-4a64-ab0c-fe6a8733a78e)

# SOLUTION

So a dirty solution is to install UnoCSS extension https://marketplace.visualstudio.com/items?itemName=antfu.unocss and add an empty `uno.config.ts` file to the root of your repo and just put it in your global gitignore.

```ts
/**
 * This file is required for the VSCode extension.
 * https://marketplace.visualstudio.com/items?itemName=antfu.unocss
 * https://unocss.dev/integrations/vscode
 * https://github.com/unocss/unocss/discussions/4257
 */
export default {}
```

The downside is that you will get double hints now and it won't work with custom definitions, only the base tailwind, otherwise you would have to implment the `uno.config.ts` file.

Tailwind only (UnoCSS not installed)
![image](https://github.com/user-attachments/assets/2654cbef-afff-4fa4-ac64-133d06dbe65e)
UnoCSS only (notice that it is faster than tailwind extension)
![image](https://github.com/user-attachments/assets/319d1c18-25a9-4ff1-b26e-35fef90575a5)
Both
![image](https://github.com/user-attachments/assets/adeee17c-5a47-47bd-accd-3bd100c075d8)
